src/clr-angular/utils/drag-and-drop/providers/drag-and-drop-event-bus.spec.ts

```diff
@@ -4,50 +4,60 @@
-                expect(value).toBe(testPayLoad);
-                expect(value.dragDataStore).toBe("dragStartTest");
+        it("should broadcast from correct Observable on drag move event type", function() {
+            this.dndEventBus.dragMoved.subscribe((event: ClrDragEvent) => {
+                isEmitted = true;
+                expect(event.type).toBe(ClrDragEventType.DRAG_MOVE);
             });
-            this.dndEventBus.notifyAllOnDragStart(testPayLoad);
+            this.dndEventBus.broadcast({type: ClrDragEventType.DRAG_MOVE});
+            expect(isEmitted).toBeTruthy();
         });
 
-        it("notifyAllOnDragMove() should cast to dragMove subject", function() {
-            const testPayLoad: ClrDraggableEvent = {dragDataStore: "dragStartMove"};
-            this.dndEventBus.dragMoved.subscribe((value: any) => {
-                expect(value).toBe(testPayLoad);
-                expect(value.dragDataStore).toBe("dragStartMove");
+        it("should broadcast from correct Observable on drag end event type", function() {
+            this.dndEventBus.dragEnded.subscribe((event: ClrDragEvent) => {
+                isEmitted = true;
+                expect(event.type).toBe(ClrDragEventType.DRAG_END);
             });
-            this.dndEventBus.notifyAllOnDragMove(testPayLoad);
+            this.dndEventBus.broadcast({type: ClrDragEventType.DRAG_END});
+            expect(isEmitted).toBeTruthy();
         });
 
-        it("notifyAllOnDragEnd() should cast to dragEnd subject", function() {
-            const testPayLoad: ClrDraggableEvent = {dragDataStore: "dragStartEnd"};
-            this.dndEventBus.dragEnded.subscribe((value: any) => {
-                expect(value).toBe(testPayLoad);
-                expect(value.dragDataStore).toBe("dragStartEnd");
+        it("should broadcast from correct Observable on drop event type", function() {
+            this.dndEventBus.dropped.subscribe((event: ClrDragEvent) => {
+                isEmitted = true;
+                expect(event.type).toBe(ClrDragEventType.DROP);
             });
-            this.dndEventBus.notifyAllOnDragEnd(testPayLoad);
+            this.dndEventBus.broadcast({type: ClrDragEventType.DROP});
+            expect(isEmitted).toBeTruthy();
         });
 
-        it("notifyAllOnDrop() should cast to drop subject", function() {
-            const testPayLoad: ClrDraggableEvent = {dragDataStore: "drop"};
-            this.dndEventBus.dropped.subscribe((value: any) => {
-                expect(value).toBe(testPayLoad);
-                expect(value.dragDataStore).toBe("drop");
+        it("should not broadcast anything on wrong event type", function() {
+            this.dndEventBus.dropped.subscribe((event: ClrDragEvent) => {
+                isEmitted = true;
             });
-            this.dndEventBus.notifyAllOnDrop(testPayLoad);
+            this.dndEventBus.broadcast({type: "if ever wrong type is passed"});
```

Comment:
Wait, how does that even compile? Typescript should throw an error here. 😮

Reply:
Yes 😂 that's why I was asking whether I should throw an exception or not. In the test files, it's compiling. But when editing other files in IDE, it notifies me by showing that squiggly error on the type.

Reply:
doesnt this mean that something is wrong with our `tslint` config?

Reply:
It's just that `this` isn't typed. But there isn't a real point to using the user context from Jasmine here, so we should just use simple variables, that way they'll be properly typed.